File: server/FlightLog-server/src/main/java/bitc/fullstack503/flightlog/flightlogserver/dto/TicketHolderDTO.java
package bitc.fullstack503.flightlog.flightlogserver.dto;


import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class TicketHolderDTO {

  @JsonProperty("flightId")
  private Integer flightId;   // 비행편

  @JsonProperty("flightStartId")
  private Integer flightStartId;

  @JsonProperty("flightArrId")
  private Integer flightArrId;

  @JsonProperty("flightPassport")
  private String flightPassport;       // 여권번호

  @JsonProperty("flightUserId")
  private String flightUserId;     // 그룹명

  @JsonProperty("flightMemFirstName")
  private String flightMemFirstName;  // 이름

  @JsonProperty("flightMemLastName")
  private String flightMemLastName;  // 성

  @JsonProperty("flightReno")
  private String flightReno;         // 예약번호

  @JsonProperty("flightReserveSDate")
  private String flightReserveSDate;      // 출발 날짜

  @JsonProperty("flightInfoStartCity")
  private String flightInfoStartCity;      // 출발 도시

  @JsonProperty("flightInfoStartTime")
  private String flightInfoStartTime;      // 출발 시간

  @JsonProperty("flightInfoArrivalCity")
  private String flightInfoArrivalCity;    // 도착 도시

  @JsonProperty("flightInfoArrivalTime")
  private String flightInfoArrivalTime;    // 도착 시간

  @JsonProperty("flightInfoAirline")
  private String flightInfoAirline;      //항공사 이름

  @JsonProperty("flightMemStartSeatNum")
  private String flightMemStartSeatNum;  //출발 좌석 번호

  @JsonProperty("flightMemArriveSeatNum")
  private String flightMemArriveSeatNum; //도착 좌석 번호

  @JsonProperty("selectedSeat")
  private String selectedSeat; // 선택된 좌석

  @JsonProperty("flightMemStartPrice")
  private Integer flightMemStartPrice; // 출발편 가격

  @JsonProperty("flightMemArrPrice")
  private Integer flightMemArrPrice; // 도착편 가격

  @JsonProperty("selectedPrice")
  private Integer selectedPrice; // 선택된 좌석 가격

  @JsonProperty("flightDistance")
  private Double flightDistance;    //비행거리

  @JsonProperty("flightMemLuggage")
  private String flightMemLuggage;     // 수하물 정보

  @JsonProperty("flightArrPayCheck")
  private String flightArrPayCheck;    // 오는편 결제 여부

  @JsonProperty("flightStartPayCheck")
  private String flightStartPayCheck; // 가는편 결제 여부

  @JsonProperty("paymentStatus")
  private String paymentStatus; // 결제 상태 추가 (예: '완료', '진행 중', '취소')
}

File: app/app/src/main/java/bitc/fullstack/FlightLog/dto/TicketHolderDTO.kt
package bitc.fullstack.FlightLog.dto

data class TicketHolderDTO(
    val flightPassport: String,          // 여권 번호
    val flightReno : String,             // 예약 번호
    val flightId: Int,                   // 비행기 ID
    val flightStartId : Int,
    val flightArrId : Int,
    val flightUserId: String,            // 티켓팅한 ID
    val flightMemFirstName:String,       // 이름
    val flightMemLastName:String,        // 성
    val flightReserveSDate: String,      // 출발 날짜
    val flightInfoAirline: String,       // 항공사 이름
    val flightInfoStartCity: String,     // 출발 도시
    val flightInfoStartTime: String,     // 출발 시간
    val flightInfoArrivalCity: String,   // 도착 도시
    val flightInfoArrivalTime: String,   // 도착 시간
    val flightMemStartSeatNum : String,  // 가는 좌석
    val flightMemArriveSeatNum : String, // 오는 좌석
    val selectedSeat : String,           // 선택된 좌석 (가는편 오는편 판단 결과 좌석)
    val flightMemStartPrice : Int,       // 가는 가격
    val flightMemArrPrice : Int,         // 오는 가격
    val selectedPrice : Int,             // 선택된 가격
    val flightMemLuggage: String,        // 수하물 정보
    val flightDistance:  Double,         // 비행 거리(가격 계산용)
    val flightArrPayCheck: String,       // 오는 날 결제 여부
    val flightStartPayCheck: String,     // 가는 날 결제 여부
    var isChecked: Boolean = false,      // 티켓 체크 상태 확인
    var paymentStatus: String? = null,    // 결제 상태 (예: '완료', '진행 중')
    val flightIds: List<Int> = listOf()  // 여러 개의 flightId를 담을 리스트 추가

)
